### PR TITLE
Re-enable cross version test for Java 9+

### PR DESCRIPTION
**Pull request by blindpirate**
### Context

See https://github.com/gradle/gradle/issues/2116 for more context. TL;DR:

Once upon a time we disabled cross version tests on Java 9 because at that time we were not ready for Java 9. Now it's time to re-enable them.

In addition, this PR fixed a little issue of `JAVA_ILLEGAL_ACCESS_WARNING_PATTERN` - a newline character was missing in this pattern.

### CODE REVIEW

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/LogContent.java [hunk 1]
```diff
@@ -39,7 +39,7 @@
     private final static String DOWN_MOVEMENT_WITH_NEW_LINE_PATTERN = "\u001b\\[\\d+B\\n";
     private final static Pattern WORK_IN_PROGRESS_AREA_PATTERN = Pattern.compile(PROGRESS_BAR_PATTERN + "|" + WORK_IN_PROGRESS_PATTERN + "|" + DOWN_MOVEMENT_WITH_NEW_LINE_PATTERN);
     private final static Pattern JAVA_ILLEGAL_ACCESS_WARNING_PATTERN = Pattern.compile("(?ms)WARNING: An illegal reflective access operation has occurred$.+?"
-        + "^WARNING: All illegal access operations will be denied in a future release$");
+        + "^WARNING: All illegal access operations will be denied in a future release\n$");
```
Comment (marcphilipp) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/LogContent.java]: Why is this change necessary? Do we need to remove an additional line? If so, we should use `^` to keep it independent of the actual line separator.

Comment (blindpirate) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/LogContent.java]: Yes, the output is `WARNING: xxxx\n` and the original regex can't remove the last newline char.

Comment (blindpirate) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/LogContent.java]: What do you mean using `^` to remove the new line? Please tell me.

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/LogContent.java [hunk 2]
```diff
@@ -39,7 +39,7 @@
     private final static String DOWN_MOVEMENT_WITH_NEW_LINE_PATTERN = "\u001b\\[\\d+B\\n";
     private final static Pattern WORK_IN_PROGRESS_AREA_PATTERN = Pattern.compile(PROGRESS_BAR_PATTERN + "|" + WORK_IN_PROGRESS_PATTERN + "|" + DOWN_MOVEMENT_WITH_NEW_LINE_PATTERN);
     private final static Pattern JAVA_ILLEGAL_ACCESS_WARNING_PATTERN = Pattern.compile("(?ms)WARNING: An illegal reflective access operation has occurred$.+?"
-        + "^WARNING: All illegal access operations will be denied in a future release$");
+        + "^WARNING: All illegal access operations will be denied in a future release\r?\n");
```
Comment (marcphilipp) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/LogContent.java]: See https://github.com/gradle/gradle/pull/5517/files#r190797297

Comment (blindpirate) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/LogContent.java]: Thanks a lot @marcphilipp !
